### PR TITLE
add stub to modules used in raredisease pipeline

### PR DESCRIPTION
**Pull request by ramprasadn**
This PR adds stub section to the following modules

- bcftools/norm
- bcftools/view
- bwamem2/index
- bwamem2/mem
- cat/cat
- deepvariant
- expansionhunter
- fastqc
- gatk4/bedtointervallist
- gatk4/createsequencedictionary
- gatk4/intervallisttools
- glnexus
- mosdepth
- multiqc
- picard/collecthsmetrics
- picard/collectmultiplemetrics
- picard/markduplicates
- picard/sortvcf
- qualimap/bamqc
- samtools/faidx
- samtools/index
- samtools/merge
- samtools/sort
- samtools/stats
- svdb/query
- tabix/bgziptabix
- tabix/tabix
- tiddit/cov
- tiddit/sv
- ucsc/wigtobigwig
- untar
- vcfanno

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by ramprasadn** [comment]
Four tests have failed apart from markdown lint. Of those four, deepvariant's conda run fails because it lacks a conda package. 
Remaining three (picard's collecthsmetrics, collectmultiplemetrics and markduplicates) fail their corresponding conda tests, but those same tests pass in a local environment. In addition to that, for these picard modules, the docker and singularity runs pass without any issues. So I think the failed test due to issues unrelated to this PR.

**Comment by ramprasadn** [comment]
@maxulysse @JoseEspinosa @drpatelh what do you guys think about this PR? 😅 

(With every new CI run, different conda runs are failing so I think those might be due to issues unrelated to this PR.)

**Review by drpatelh — APPROVED** [review]
Awesome work @ramprasadn ! Thank you!

Happy for this to be merged. @maxulysse you want to have a final look?

File: modules/bcftools/view/main.nf [hunk 1]
```diff
@@ -41,4 +41,14 @@ process BCFTOOLS_VIEW {
         bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.vcf.gz
```
Comment (JoseEspinosa) [inline modules/bcftools/view/main.nf:48]:
```suggestion
    touch ${prefix}.vcf.gz
 
```

File: modules/bcftools/norm/main.nf [hunk 2]
```diff
@@ -34,4 +34,14 @@ process BCFTOOLS_NORM {
         bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.vcf.gz
```
Comment (JoseEspinosa) [inline modules/bcftools/norm/main.nf:41]:
```suggestion
    touch ${prefix}.vcf.gz
    
```

File: modules/bwamem2/index/main.nf [hunk 3]
```diff
@@ -31,4 +31,18 @@ process BWAMEM2_INDEX {
...
     """
+
+    stub:
+    """
+    mkdir bwamem2
+    touch bwamem2/${fasta}.0123
+    touch bwamem2/${fasta}.ann
+    touch bwamem2/${fasta}.pac
+    touch bwamem2/${fasta}.amb
+    touch bwamem2/${fasta}.bwt.2bit.64
```
Comment (JoseEspinosa) [inline modules/bwamem2/index/main.nf:42]:
```suggestion
    touch bwamem2/${fasta}.bwt.2bit.64
    
```

File: modules/expansionhunter/main.nf [hunk 4]
```diff
@@ -37,4 +37,14 @@ process EXPANSIONHUNTER {
         expansionhunter: \$( echo \$(ExpansionHunter --version 2>&1) | sed 's/^.*ExpansionHunter v//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.vcf
```
Comment (JoseEspinosa) [inline modules/expansionhunter/main.nf:44]:
```suggestion
    touch ${prefix}.vcf
    
```

File: modules/gatk4/intervallisttools/main.nf [hunk 5]
```diff
@@ -51,4 +51,15 @@ process GATK4_INTERVALLISTTOOLS {
         gatk4: \$(echo \$(gatk --version 2>&1) | sed 's/^.*(GATK) v//; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    mkdir ${prefix}_split
```
Comment (JoseEspinosa) [inline modules/gatk4/intervallisttools/main.nf:58]: Have you checked that just creating the folder works, the output channel is declared in this manner:
```
 tuple val(meta), path("*_split/*/*.interval_list"), emit: interval_list
 ```
 so I would say that should maintain the same structure

File: modules/multiqc/main.nf [hunk 6]
```diff
@@ -28,4 +28,15 @@ process MULTIQC {
         multiqc: \$( multiqc --version | sed -e "s/multiqc, version //g" )
     END_VERSIONS
     """
+
+    stub:
+    """
+    touch multiqc_data
+    touch multiqc_plots
+    touch multiqc_report.html
```
Comment (JoseEspinosa) [inline modules/multiqc/main.nf:36]:
```suggestion
    touch multiqc_report.html

```

File: modules/picard/markduplicates/main.nf [hunk 7]
```diff
@@ -42,4 +42,16 @@ process PICARD_MARKDUPLICATES {
         picard: \$(echo \$(picard MarkDuplicates --version 2>&1) | grep -o 'Version:.*' | cut -f2- -d:)
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.bam
+    touch ${prefix}.bam.bai
+    touch ${prefix}.MarkDuplicates.metrics.txt
```
Comment (JoseEspinosa) [inline modules/picard/markduplicates/main.nf:51]:
```suggestion
    touch ${prefix}.MarkDuplicates.metrics.txt

```

File: modules/picard/sortvcf/main.nf [hunk 8]
```diff
@@ -46,4 +46,16 @@ process PICARD_SORTVCF {
         picard: \$(picard SortVcf --version 2>&1 | grep -o 'Version:.*' | cut -f2- -d:)
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}_sorted.vcf.gz
+    touch ${prefix}.bam.bai
+    touch ${prefix}.MarkDuplicates.metrics.txt
```
Comment (JoseEspinosa) [inline modules/picard/sortvcf/main.nf:55]:
```suggestion
    touch ${prefix}_sorted.vcf.gz

```

File: modules/qualimap/bamqc/main.nf [hunk 9]
```diff
@@ -52,4 +52,71 @@ process QUALIMAP_BAMQC {
...
+    touch mapped_reads_clipping_profile.txt
+    touch mapping_quality_across_reference.txt
+    touch duplication_rate_histogram.txt
+    touch insert_size_across_reference.txt
+    touch mapped_reads_gc-content_distribution.txt
+    touch mapping_quality_histogram.txt
+    cd ../
+    touch genome_results.txt
+    touch qualimapReport.html
+    cd ../
```
Comment (JoseEspinosa) [inline modules/qualimap/bamqc/main.nf:116]:
```suggestion
    cd ../

```

File: modules/samtools/faidx/main.nf [hunk 10]
```diff
@@ -29,4 +29,13 @@ process SAMTOOLS_FAIDX {
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    """
+    touch ${fasta}.fai
```
Comment (JoseEspinosa) [inline modules/samtools/faidx/main.nf:35]:
```suggestion
    touch ${fasta}.fai

```

File: modules/samtools/index/main.nf [hunk 11]
```diff
@@ -33,4 +33,15 @@ process SAMTOOLS_INDEX {
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    """
+    touch ${input}.bai
+    touch ${input}.crai
+    touch ${input}.csi
```
Comment (JoseEspinosa) [inline modules/samtools/index/main.nf:41]:
```suggestion
    touch ${input}.csi

```

File: modules/samtools/merge/main.nf [hunk 12]
```diff
@@ -38,4 +38,15 @@ process SAMTOOLS_MERGE {
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
+    def file_type = input_files[0].getExtension()
+    """
+    touch ${prefix}.${file_type}
```
Comment (JoseEspinosa) [inline modules/samtools/merge/main.nf:46]:
```suggestion
    touch ${prefix}.${file_type}

```

File: modules/samtools/sort/main.nf [hunk 13]
```diff
@@ -28,4 +28,14 @@ process SAMTOOLS_SORT {
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.bam
```
Comment (JoseEspinosa) [inline modules/samtools/sort/main.nf:35]:
```suggestion
    touch ${prefix}.bam

```

File: modules/samtools/stats/main.nf [hunk 14]
```diff
@@ -34,4 +34,14 @@ process SAMTOOLS_STATS {
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${input}.stats
```
Comment (JoseEspinosa) [inline modules/samtools/stats/main.nf:41]:
```suggestion
    touch ${input}.stats
 
```

File: modules/tabix/bgziptabix/main.nf [hunk 15]
```diff
@@ -30,4 +30,15 @@ process TABIX_BGZIPTABIX {
         tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.gz
+    touch ${prefix}.gz.tbi
```
Comment (JoseEspinosa) [inline modules/tabix/bgziptabix/main.nf:38]:
```suggestion
    touch ${prefix}.gz.tbi

```

File: modules/svdb/query/main.nf [hunk 16]
```diff
@@ -63,12 +63,11 @@ process SVDB_QUERY {
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}_query.vcf
+    touch ${prefix}_ann_svdbq.vcf
```
Comment (JoseEspinosa) [inline modules/svdb/query/main.nf:66]: Why this change the output channel declares `*_query.vcf` right?

File: modules/svdb/query/main.nf [hunk 17]
```diff
@@ -63,12 +63,11 @@ process SVDB_QUERY {
...
 
     cat <<-END_VERSIONS > versions.yml
```
Comment (JoseEspinosa) [inline modules/svdb/query/main.nf:68]:
```suggestion

    cat <<-END_VERSIONS > versions.yml
```

File: modules/tabix/tabix/main.nf [hunk 18]
```diff
@@ -27,4 +27,14 @@ process TABIX_TABIX {
         tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${tab}.tbi
```
Comment (JoseEspinosa) [inline modules/tabix/tabix/main.nf:34]:
```suggestion
    touch ${tab}.tbi

```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Very nice! 😄 
Added a pair of suggestions about the naming of the files where I think there might be a problem, all the other suggestions are just related to adding an empty line just before the `versions.yml` creation for consistency.

File: modules/svdb/query/main.nf (hunk 16 above)
Comment (ramprasadn) [reply modules/svdb/query/main.nf]: Good catch! Fixed.

File: modules/gatk4/intervallisttools/main.nf (hunk 5 above)
Comment (ramprasadn) [reply modules/gatk4/intervallisttools/main.nf]: I agree! I have fixed it now.

Comment (JoseEspinosa) [reply modules/gatk4/intervallisttools/main.nf]: 🚀
